### ISSUE TITLE
bug(menu): focusOnOpen has no default value

### ISSUE THREAD

**Issue by mgr34** [jamesmfriedman/rmwc#579]
on the 6.0.0-alpha branches The documentation for `<Menu/>` notes that `focusOnOpen` defaults to `true`.  This is not the case, it defaults to undefined. 

I'd like to submit a PR, but I am a bit confused how to set a `defaultValue`. I've mentioned this in discord if you prefer to have the discussion there. I'm only so familiar with TS, but in this case, its the handy `createComponent` that is throwing me off. 

I could change the foundation `useEffect` hook to look for props.focsOnOpen to be `undefined || true` but imagine setting a true defaultValue maybe preferred.

**Comment by jamesmfriedman** [comment]
Either place is fine. You are correct that the default prop for focusOnOpen was lost in the transition to hooks.

As a whole, I don't think defaultProps are used anywhere in the library anymore since you can set individual props using destructuring. 

The issue is located here:
https://github.com/jamesmfriedman/rmwc/blob/4463d1a3f36ce9d2c35aee8b4eb0df912277433b/src/menu/menu-foundation.tsx#L99

I know personally I have relied on implicit behavior instead of setting a default in a few places, but I can't say I have a real preference. The type issue you're referring to is likely that you're not allowed to declare defaultProps on a component created from the  `createComponent` helper.  That is something that can also be fixed.

Open to your opinion on how you want to fix it, and I can help either way.

**Comment by mgr34** [comment]
Thanks for the quick response.

you are right it is a `useCallback` not `useEffect`. My bad.

If I am understanding correctly there is not currently a way to set default prop values when using `createComponent`?  

In that case I think I will just change 
```js
 if (
        props.focusOnOpen &&
        list.length > 0 &&
        !list.some(el => el === document.activeElement)
      ) {}
```

to 
```js
 if (
        (props.focusOnOpen || props.focusOnOpen === undefined) &&
        list.length > 0 &&
        !list.some(el => el === document.activeElement)
      ) {}
```

**Comment by jamesmfriedman** [comment]
Fixed in 6.0.0
